flutter/lib/ui/geometry.dart

```diff
@@ -837,39 +982,47 @@ class RRect {
   }
 
   /// The point halfway between the left and right and the top and bottom edges of this rectangle.
-  Point get center => new Point(left + width / 2.0, top + height / 2.0);
+  Point get center => new Point(left + width / 2, top + height / 2);
 
   /// Whether the given point lies inside the rounded rectangle.
   bool contains(Point point) {
     if (point.x < left || point.x >= right || point.y < top || point.y >= bottom)
       return false; // outside bounding box
-    double leftInner = left + radiusX;
-    double rightInner = right - radiusX;
-    double topInner = top + radiusY;
-    double bottomInner = bottom - radiusY;
-    if (point.x >= leftInner && point.x <= rightInner)
-      return true; // inside tallMiddleRect
-    if (point.y >= topInner && point.y <= bottomInner)
-      return true; // inside wideMiddleRect
-    // it is in one of the corners
-    // convert this to a test of the unit circle
-    double x, y;
-    if (point.x > leftInner) {
-      assert(point.x > rightInner);
-      x = point.x - (rightInner - leftInner);
-    } else {
-      x = point.x;
-    }
-    if (point.y > topInner) {
-      assert(point.y > bottomInner);
-      y = point.y - (bottomInner - topInner);
+
+    double x;
+    double y;
+    double radiusX;
+    double radiusY;
+    // check whether point is in one of the rounded corner areas
+    // x, y -> translate to ellipse center
+    if (point.x < left + tlRadiusX && point.y < top + tlRadiusY) {
+      x = point.x - left - tlRadiusX;
+      y = point.y - top - tlRadiusY;
+      radiusX = tlRadiusX;
+      radiusY = tlRadiusY;
+    } else if (point.x > right - trRadiusX && point.y < top + trRadiusY) {
+      x = point.x - right + trRadiusX;
+      y = point.y - top - trRadiusY;
+      radiusX = trRadiusX;
+      radiusY = trRadiusY;
+    } else if (point.x > right - brRadiusX && point.y > bottom - brRadiusY) {
+      x = point.x - right + brRadiusX;
+      y = point.y - bottom + brRadiusY;
+      radiusX = brRadiusX;
+      radiusY = brRadiusY;
+    } else if (point.x < left + blRadiusX && point.y > bottom - blRadiusY) {
+      x = point.x - left - blRadiusX;
+      y = point.y - bottom + blRadiusY;
+      radiusX = blRadiusX;
+      radiusY = blRadiusY;
     } else {
-      y = point.y;
+      return true; // inside and not within the rounded corner area
     }
-    x = x / (radiusX * 2) - 0.5;
-    y = y / (radiusY * 2) - 0.5;
+
+    x = x / radiusX;
+    y = y / radiusY;
     // check if the point is outside the unit circle
-    if (x * x + y * y > 0.25)
+    if (x * x + y * y > 1.0)
```

Comment:
you should write a test for this and add it on the framework side (we don't have tests on the engine side currently).